sklearn/decomposition/pca.py

```diff
@@ -216,6 +217,77 @@ def __init__(self, n_components=None, copy=True, whiten=False):
         self.copy = copy
         self.whiten = whiten
 
+    def get_feature_names(self, input_features=None, full_output=False):
+        """
+        Return dominant feature names for each component
+
+        Parameters
+        ----------
+        input_features : list of string, shape (n_features), optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        full_output : boolean or integer, default False
+            When it is "True", return the principal components as the
+            combination of the input features. If "False", will be just
+            the component names. If it is an integer n, it returns the
+            sorted top n contributions to each component.
+
+
+        Returns
+        -------
+        output_feature_names : list of string, shape (n_components)
+            When full_output is "True", it is represented by the contribution
+            of input features and full_output is "False", it just represents
+            the component names
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from sklearn.decomposition import PCA
+        >>> X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
+        >>> pca = PCA(n_components=2).fit(X)
+        >>> pca.get_feature_names(full_output=True)
+        ['0.84*x0 + 0.54*x1', '0.54*x0 - 0.84*x1']
+        >>> pca.get_feature_names(full_output=1)
+        ['0.84*x0', '-0.84*x1']
+        >>> pca.get_feature_names()
+        ['pc0', 'pc1']
+        """
+        check_is_fitted(self, 'components_')
+
+        n_features = self.components_.shape[1]
+        components = self.components_
+        contribution = np.argsort(-np.abs(components), axis=1)
```

Comment:
well, `np.argsort(np.abs(components), axis=1)[::-1]` is cheaper, but only marginally so.